index.js

```diff
@@ -24,6 +24,15 @@ async function transition(direction, element, animation) {
     const start = dataset[`${transition}Start`] ? dataset[`${transition}Start`].split(" ") : [`${animationClass}-start`]
     const end = dataset[`${transition}End`] ? dataset[`${transition}End`].split(" ") : [`${animationClass}-end`]
 
+    // if there's any overlap between the current set of classes and genesis/start/end,
+    // we should remove them before we start and add them back at the end
+    const stash = []
+    const current = new Set(element.classList.values())
+    genesis.forEach(cls => current.has(cls) && stash.push(cls))
+    start.forEach(cls => current.has(cls) && stash.push(cls))
+    end.forEach(cls => current.has(cls) && stash.push(cls))
```

Comment:
I think this can be simplified. 

```suggestion
  const elementClassList = new Set(element.classList.values())
  const intersect = [...genesis, ...start, ...end].filter(c => elementClassList.has(c))
```